solutions/LC Find Pivot Index.mdx

```diff
@@ -0,0 +1,192 @@
+---
+id: lc-find-pivot-index
+source: LeetCode
+title: Find Pivot Index
+author: Qi Wang
+---
+
+**Time Complexity:** $\mathcal O(N)$.
+
+## Solution 1
+
+The pivot index must have the same sum on its left and its right. To calculate this, we can create a prefix sum of the $\texttt{nums}$ array (in my solution it is cushioned with one $0$ at index $0$). Afterwards, all we need to do is to loop through every index from $0$ (inclusive) to $\texttt{nums}.length$ (exclusive) in the $\texttt{psums}$ array and check for:
+
+$$
+\texttt{psums}[i] - \texttt{psums}[0] == \texttt{psums}[\texttt{nums}.length] - \texttt{psums}[i+1]
+$$
+
+If this is true, we can return our current index because the problem wants the left-most index. If it finishes the for-loop without returning, it means there is no valid pivot index so return $-1$.
+
+## Solution 2
+
+The alternate solution does not require an array to store the sums. Instead, we only need the total sum of the array and the current sum at each index (exclusive of index). With this we check for:
+
+$$
+\texttt{curSum} == \texttt{total} - \texttt{curSum} - \texttt{nums}[i];
+$$
+
+If this is true, we can return our current index because the problem wants the left-most index. If it finishes the for-loop without returning, it means there is no valid pivot index so return $-1$.
```

Comment:
> If this is true, we can return our current index because the problem wants the left-most index. If it finishes the for-loop without returning, it means there is no valid pivot index so return $-1$.

You can remove this part as you stated it in sol.1